enhancements/cloud-integration/tokenized-auth-enablement-operators-on-cloud.md

```diff
@@ -0,0 +1,404 @@
+// CredentialsRequest.
+// The presence of an STSRoleARN within the AWSProviderSpec initiates creation of a secret containing IAM
+// Role details necessary for assuming the IAM Role via Amazon's Secure Token Service.
+// +optional
+STSIAMRoleARN string `json:"stsIAMRoleARN,omitempty"`
+}
+```
+
+On AWS STS, the `STSIAMRoleARN` would be something like:
+`arn:aws:iam::269733383066:role/newstscluster-openshift-logging-role-name`<br>
+This provides the role information linked by policy to access various cloud resources.
+
+`CloudTokenPath` provides the path to a JWT web token needed to access the resources.
+
+### Risks and Mitigations
+
+### Drawbacks
+
+## Design Details
+
+We have two PoC items:<br>
+How CCO changes might be implemented: https://github.com/openshift/cloud-credential-operator/pull/525
+
+How Cluster Logging Operator can be changed to work with this modified CCO: 
+https://github.com/gallettilance/cluster-logging-operator/tree/sts
+
+### Test Plan
+
+Consider the following in developing a test plan for this enhancement:
+- CCO should add unit tests for the new manner of working when in Manual mode but the 3-way test indicates running on
+  a TAT enabled cluster.
+- CCO should add e2e tests for the new manner of working when in Manual mode but the 3-way test indicates running on
+  a TAT enabled cluster.
+- CI will need to add a TAT mode for various cloud providers as a base for the e2e tests.
+- A "canary" operator should be fielded in CI with e2e tests such that we have signal on breaking changes in TAT enabled
+  clusters.
+
+### Graduation Criteria
+
+#### Tech Preview -> GA
+This feature will be introduced as part of the TechPreviewNoUpgrade featureset affecting the Cloud Credential Operator, Console, and OLM. It is expected to graduate to GA within the same release upon successfully demonstrating e2e test pass rate meets or exceeds OCP baselines.
+
+If the feature remains `TechPreviewNoUpgrade` OLM operators which had been updated assuming this feature is available should continue to provide admin documentation sufficient to manually enable TAT.
+#### Removing a deprecated feature
+#### Support Procedures
+
+### Upgrade / Downgrade Strategy
+
+### Version Skew Strategy
+
+### Operational Aspects of API Extensions
+
+Should be none. If the new fields on CredentialsRequest are missing, cloud resources will be unavailable for an 
+operator, but there should be no other impacts.
+
+#### Failure Modes
+
+If the Secrets created by CCO on the added CredentialsRequest fields contemplated in this EP are missing, operators should wait the Secrets to appear. If
+after a timeout period, the fields are not present, the operator should write events about the lack of access to cloud 
+resources to the status of an appropriate resource to let admins know there are missing credentials.
+
+## Implementation History
+
+Not implemented yet.
+
+## Alternatives
```

Comment:
I'm curious has there been any consideration for the CCO to be consumed as a dependency of the OLM operator requiring it?

Reply:
@enxebre Did you mean that a CCO like operator could be delivered via Operator Hub and operators which require this feature could depend on that operator?